contrib/IECoreUSD/src/IECoreUSD/PointInstancerAlgo.cpp

```diff
@@ -0,0 +1,139 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2020, Cinesite VFX Ltd. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "DataAlgo.h"
+#include "ObjectAlgo.h"
+#include "PrimitiveAlgo.h"
+
+#include "IECoreScene/PointsPrimitive.h"
+
+IECORE_PUSH_DEFAULT_VISIBILITY
+#include "pxr/usd/usdGeom/pointInstancer.h"
+IECORE_POP_DEFAULT_VISIBILITY
+
+using namespace IECore;
+using namespace IECoreScene;
+using namespace IECoreUSD;
+
+//////////////////////////////////////////////////////////////////////////
+// Reading
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+IECore::ObjectPtr readPointInstancer( pxr::UsdGeomPointInstancer &pointInstancer, pxr::UsdTimeCode time )
+{
+	pxr::VtVec3fArray pointsData;
+	pointInstancer.GetPositionsAttr().Get( &pointsData, time );
+	IECore::V3fVectorDataPtr positionData = DataAlgo::fromUSD( pointsData );
+	positionData->setInterpretation( GeometricData::Point );
+	IECoreScene::PointsPrimitivePtr newPoints = new IECoreScene::PointsPrimitive( positionData );
```

Comment:
Are we going to need some way of knowing this came from a `UsdGeomPointInstancer` rather than `UsdGeomPoints` when we deal with write, along with a way to turn an `IECore::PointsPrimitive` into a `USDGeomPointInstancer`?  Maybe its enough to assume that if it has the required primvars then thats the intention?

Reply:
Round-tripping UsdGeomPointInstancers isn't on my list of things to deal with right now. My assumption is that the things listed in the Gaffer tracker take precedence.